compiler-plugin/src/test/kotlin/arrow/meta/plugins/proofs/ResolutionTests.kt

```diff
@@ -0,0 +1,397 @@
+package arrow.meta.plugins.proofs
+
+import arrow.meta.plugin.testing.Assert
+import arrow.meta.plugin.testing.CompilerTest
+import arrow.meta.plugin.testing.Dependency
+import arrow.meta.plugin.testing.assertThis
+import arrow.meta.plugins.refinement.RefinementTests
+import arrow.meta.plugins.typeclasses.GivenTest
+import org.junit.Ignore
+import org.junit.Test
+
+// TODO: build ide peace with annotator
+class ResolutionTests {
+  @Test
+  fun `prohibited public proof of non user types`() {
+    resolutionTest(
+      """
+      @Coercion
+      fun String.toInt16(): Int? =
+        toIntOrNull(16)
+        
+      val x: Int? = "30"
+      """) {
+      failsWith {
+        it.contains("This ExtensionProof test.toInt16: kotlin.String -> kotlin.Int? violates ownership rules, because public Proofs over 3rd party Types break coherence over the kotlin ecosystem. One way to solve this is to declare the Proof as an internal orphan.")
```

Comment:
I think we can drop the whole breaks coherence and provide a link instead to the docs of meta where this is explained. Otherwise, we should just simply render the signature of the function with the `internal` keyword for users to just replace it with the suggestion or copy and paste the suggestion from their terminal

Reply:
Sure, we can leave a TODO when the link to the Docs exists. 
I'll add an Issue with Arrow Error messages, just like we have with https://github.com/arrow-kt/arrow-meta/issues/131#issuecomment-555540571